hw/ip/test/src/tb_lib.hh

```diff
@@ -121,7 +121,12 @@ struct BootData {
     uint32_t core_count;
     uint32_t hartid_base;
     uint32_t tcdm_start;
-    uint32_t tcdm_end;
+    uint32_t tcdm_size;
+    uint32_t tcdm_offset;
```

Comment:
I was just wondering whether we want to make this an array and individual for each cluster e.g.,
```
uint32_t tcdm_addr [NR_CLUSTERS];
```

Because technically the TCDMs can be arbitrarily spaced. What do you think?

Reply:
Could we auto-generate the `bootdata` somehow in that case? Otherwise this gets a bit tedious when switching between different configurations

Reply:
Ah I see, we might want to make this a pointer so that we are not defining the size in the header and populate it correctly (using a script) in the `bootdata.cc`. Does that make sense?

I think for Occamy the `bootdata.cc` has been hard-coded so far, but for the single cluster version we should already be generating it from a script so we should extend that approach. I think that will become more and more useful, especially if we start to describe more complex systems.

Reply:
> Ah I see, we might want to make this a pointer so that we are not defining the size in the header and populate it correctly (using a script) in the `bootdata.cc`. Does that make sense?

Not quite, this `bootdata` struct is defined in three places `tb_lib.hh`, `start_occamy.c` and `bootdata.cc`. As far as I understand, those need to match

Reply:
As discussed in the daily today. @noah95 will tackle that in the next weeks. We are good for now.